### DISCUSSION TITLE
Why did you disable Fuse?

### DISCUSSION THREAD

**Discussion by bitshiftnetau** [Homebrew/discussions#3155] [Everyday usage] [answered]
Fuse is needed to read ext4 on osx... why would you disable something because it's closed-source when it's being installed on a closed-source ecosystem? I'm clearly missing something here...

**Comment by SMillerDev** [comment]
Because it does not meet our requirements: https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae

**Reply by bitshiftnetau** [reply]
So why do you develop software for a closed-source ecosystem? I can re-enable it simply by editing the formula in vi, so like it still works and it's still there... you're just deciding not to allow it for some arbitrary reason that falls contrary to the systems you service. 

You're a package manager, a conduit for my whims and the whims of those installing packages, why would you have control over the packages or functionality that is installed from said conduit?

**Reply by SMillerDev** [reply]
> So why do you develop software for a closed-source ecosystem?

Because I want to use that system.

> I can re-enable it simply by editing the formula in vi, so like it still works and it's still there... you're just deciding not to allow it for some arbitrary reason that falls contrary to the systems you service.

Until we completely remove it it is true that you can re-enable it. But just because the system that Homebrew was created for doesn't mean we have to support all closed source software.

> You're a package manager, a conduit for my whims and the whims of those installing packages, why would you have control over the packages or functionality that is installed from said conduit?

We're a package manager so we package software. That is the only claim https://github.com/Homebrew/homebrew-core/blob/master/LICENSE.txt will allow you to make without exceeding the demands of the provided support.

**Reply by bitshiftnetau** [reply]
Sure you don't have to but why wouldn't you want to? ext4 is still a major file system and fuse seems to be the only way to achieve read-only access. 

Is there another way to transfer data from ext4 onto APFS on OSX? Honestly it seems to be a little self-defeating by all parties given how difficult it's been to do this transfer. Ext4fuse isn't even working properly, my best bet atm is to just format as HFS+, transfer across, then remount on osx and transfer across to APFS... doesn't that seem a little insane? Like, wouldn't Apple and the surrounding community want to push adoption more officially?

**Reply by SMillerDev** [reply]
> Sure you don't have to but why wouldn't you want to?

Because we these are the rules we set for ourselves and it might make things difficult with licenses. 

> Like, wouldn't Apple and the surrounding community want to push adoption more officially?

Apple has no influence in Homebrew and Homebrew has no influence in macFUSE. If Apple wants to make it easier to work with ext4 I'm sure there are ways they can do that, and if macFUSE wants Homebrew to adopt their software in the Open Source component of Homebrew then they can always offer the software under an open source license again.

**Reply by carlocab** [reply]
There are many reasons why we try to package only open source software as formulae in homebrew-core.

The (arguably) most important one is legal: many/most non-open source software have licenses which restrict their redistribution. Open source software, on the other hand, places practically no limitation on packaging them.

It thus makes perfect sense for a package manager to package only open-source software. Indeed, most widely-used package managers require an open source license before packaging a piece of software.

**Comment by gromgit** [comment, marked as answer]
> Fuse is needed to read ext4 on osx... why would you disable something because it's closed-source when it's being installed on a closed-source ecosystem? I'm clearly missing something here...

You're missing at least two things:
1. Homebrew's project, Homebrew's rules.
2. I'm maintaining the FUSE formulae for macOS in my own tap (https://github.com/gromgit/homebrew-fuse) on a best-effort basis. Some software has moved on to the FUSE v3 API, so they'll not work at all till macFUSE implements the same. I'm likely missing some formulae that were introduced into `homebrew/core` after July 2021 or so; if you need it, raise an issue on my tap (https://github.com/gromgit/homebrew-fuse/issues/new/choose) and I'll see what I can do.
